tools/serverpod_cli/lib/src/analyzer/models/validation/restrictions.dart

```diff
@@ -158,7 +159,22 @@ class Restrictions {
     if (reservedClassNames.contains(className)) {
       return [
         SourceSpanSeverityException(
-          'The class name "$className" is reserved and cannot be used.',
+          'The ${Keyword.classType} name "$className" is reserved and cannot be used.',
+          span,
+        )
+      ];
+    }
+
+    var duplicateExtraClass =
+        (config.extraClasses as List<TypeDefinition?>).firstWhere(
+      (extraClass) => extraClass?.className == className,
+      orElse: () => null,
+    );
```

Comment:
Since `collection` is not a [dependency of serverpod_cli](https://github.com/serverpod/serverpod/blob/main/tools/serverpod_cli/pubspec.yaml), this seems to be the way to do it. Not sure if we want to consider adding the `collection` dependency like we do in the main serverpod package, which would contain the `firstWhereOrNull` helper?